dace/transformation/interstate/loop_overwrite_elimination.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+"""Eliminates loop overwriting data containers"""
+
+from dace import sdfg as sd
+from dace.sdfg import utils as sdutil
+from dace.sdfg.state import ControlFlowRegion, LoopRegion, ConditionalBlock
+from dace.transformation import transformation, helpers
+from dace.transformation.passes.analysis import loop_analysis
+from dace.sdfg.sdfg import InterstateEdge
+from dace import symbolic, nodes
+from dace.subsets import intersects
+import copy
+
+
+@transformation.explicit_cf_compatible
+class LoopOverwriteElimination(transformation.MultiStateTransformation):
+    """
+    Eliminates loops which write to the same location in each iteration by replacing the loop with the last iteration.
+    """
+
+    loop = transformation.PatternNode(LoopRegion)
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.loop)]
+
+    def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
+        # Check if this is a for-loop with known range.
+        start = loop_analysis.get_init_assignment(self.loop)
+        end = loop_analysis.get_loop_end(self.loop)
+        stride = loop_analysis.get_loop_stride(self.loop)
+        if start is None or end is None or stride is None:
+            return False
+
+        # Check if any continue, break, or return statements are present in the loop.
+        if self.loop.has_break or self.loop.has_continue or self.loop.has_return:
+            return False
+
+        # Find symbols depending on the loop variable
+        sym_deps = {}
+        for edge in self.loop.all_interstate_edges():
```

Comment:
I believe this ignores symbols assigned by other loops contained inside this loop

Reply:
Fixed by considering all edges of all controlflow block inside the loop